### ISSUE TITLE
AppArmor Syntax Error: Invalid Regex @{PROC}

### ISSUE THREAD

**Issue by kravietz** [docker/docker#19494]
Any attempt to change the AppArmor docker profile (e.g. aa-complain) results in this syntax error:
```
apparmor.common.AppArmorException: 'Syntax Error: Invalid Regex @{PROC}/{*,**^[0-9*],sys/kernel/shm*} in file: /etc/apparmor.d/docker line: 16'
```
Package versions:
```
ii  apparmor                        2.8.95~2430-0ubuntu5 amd64                User-space parser utility for AppArmor
un  apparmor-docs                   <none>               <none>               (no description available)
un  apparmor-easyprof               <none>               <none>               (no description available)
un  apparmor-easyprof-ubuntu        <none>               <none>               (no description available)
un  apparmor-parser                 <none>               <none>               (no description available)
ii  apparmor-profiles               2.8.95~2430-0ubuntu5 all                  Profiles for AppArmor Security policies
ii  apparmor-utils                  2.8.95~2430-0ubuntu5 amd64                Utilities for controlling AppArmor
```
The file is re-created every time Docker service reloads so I can't just edit it. This is Ubuntu 14.04.3 LTS so pretty mainstream one.

**Comment by kravietz** [comment]
``
# docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:12:04 UTC 2015
 OS/Arch:      linux/amd64
```

**Comment by kravietz** [comment]
```
# docker info
Containers: 6
Images: 327
Server Version: 1.9.1
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 339
 Dirperm1 Supported: true
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.19.0-47-generic
Operating System: Ubuntu 14.04.3 LTS
CPUs: 8
Total Memory: 15.64 GiB
Name: worker3
ID: CUAR:XPYJ:JLNX:6OIU:GSCE:GQ2M:L4A4:EJBB:UXY5:VCJF:3DE4:PRN5
WARNING: No swap limit support
```

**Comment by kravietz** [comment]
This is a physical server. Steps to reproduce are:
```
# apt-get install apparmor-utils
...trim...
# service docker reload
# aa-status
...trim...
3 processes are in enforce mode.
   docker-default (8833) 
   docker-default (8874) 
   docker-default (8880)
# aa-complain docker-default
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apparmor/aa.py", line 2910, in parse_profile_data
    re.compile(p_re)
  File "/usr/lib/python3.4/re.py", line 223, in compile
    return _compile(pattern, flags)
  File "/usr/lib/python3.4/re.py", line 294, in _compile
    p = sre_compile.compile(pattern, flags)
  File "/usr/lib/python3.4/sre_compile.py", line 568, in compile
    p = sre_parse.parse(p, flags)
  File "/usr/lib/python3.4/sre_parse.py", line 765, in parse
    raise error("unbalanced parenthesis")
sre_constants.error: unbalanced parenthesis

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/sbin/aa-complain", line 30, in <module>
    tool.cmd_complain()
  File "/usr/lib/python3/dist-packages/apparmor/tools.py", line 154, in cmd_complain
    apparmor.read_profiles()
  File "/usr/lib/python3/dist-packages/apparmor/aa.py", line 2594, in read_profiles
    read_profile(profile_dir + '/' + file, True)
  File "/usr/lib/python3/dist-packages/apparmor/aa.py", line 2620, in read_profile
    profile_data = parse_profile_data(data, file, 0)
  File "/usr/lib/python3/dist-packages/apparmor/aa.py", line 2912, in parse_profile_data
    raise AppArmorException(_('Syntax Error: Invalid Regex %(path)s in file: %(file)s line: %(line)s') % { 'path': path, 'file': file, 'line': lineno + 1 })
apparmor.common.AppArmorException: 'Syntax Error: Invalid Regex @{PROC}/{*,**^[0-9*],sys/kernel/shm*} in file: /etc/apparmor.d/docker line: 16'
```

**Comment by kravietz** [comment]
These lines are offending:
```
# grep PROC /etc/apparmor.d/docker 
  deny @{PROC}/{*,**^[0-9*],sys/kernel/shm*} wkx,
  deny @{PROC}/sysrq-trigger rwklx,
  deny @{PROC}/mem rwklx,
  deny @{PROC}/kmem rwklx,
  deny @{PROC}/kcore rwklx,
```

**Comment by thaJeztah** [comment]
I think this has been fixed on master in this PR; https://github.com/docker/docker/pull/19069

**Comment by kravietz** [comment]
@thaJeztah this helped indeed, thanks!

**Comment by thaJeztah** [comment]
@kravietz You're welcome! I'll close this issue because this is already fixed in the repository, and will be part of 1.10, but feel free to comment here after I've closed :+1:
